src/main/java/seedu/gamebook/logic/commands/EditCommand.java

```diff
@@ -319,13 +324,23 @@ public boolean equals(Object other) {
             EditGameEntryDescriptor e = (EditGameEntryDescriptor) other;
 
             // assume different game entries must be unique in their fields
+            boolean isTagsSame = getTags().equals(e.getTags())
+                    || getTags().equals(Optional.empty()) && e.getTags().get().size() == 0
+                    || e.getTags().equals(Optional.empty()) && getTags().get().size() == 0;
```

Comment:
This is required otherwise if getTags() is `optional.empty()` and the other object's getTags() is an empty set, equals will return false, which is not the desired behavior.